File: backend/src/main/java/com/ladysparks/ttaenggrang/dto/stock/StockTransactionDTO.java
package com.ladysparks.ttaenggrang.dto.stock;

import com.ladysparks.ttaenggrang.domain.stock.TransType;
import jakarta.persistence.Column;
import lombok.Getter;
import lombok.Setter;

import java.math.BigDecimal;
import java.sql.Timestamp;

@Getter
@Setter
public class StockTransactionDTO {
    private int id;
    private int share_count;  // 주식 거래 수량
    private Timestamp trans_date;  // 거래 날짜
    private int purchase_prc;   // 거래 당시 1주 가격
    private int total_amt;  // 총 거래 금액
    private int return_amt;   // 현재 주가
    private BigDecimal returnRate;  // 손익/손실 금액
    private TransType transType; // 거래 유형

    //조인

    // 학생 관련 (student_id 외래 키를 참조)
    private Long studentId;  // student_id 외래 키 (Student 엔티티 참조)

    // 계좌 관련 (bank_account_id 외래 키를 참조)
    private Long bankAccountId; // bank_account_id 외래 키 (BankAccount 엔티티 참조)

    // 주식 관련 (stock_id 외래 키를 참조)
    private Long stockId;     // stock_id 외래 키 (Stock 엔티티 참조)



}

File: frontend/Ttaenggrang/app/src/main/java/com/ladysparks/ttaenggrang/data/model/dto/StockTransactionDto.kt
package com.ladysparks.ttaenggrang.data.model.dto

data class StockTransactionDto(
    val id: Long = 0L,
    val shareCount: Int = 0,
    val transDate: String,
    val purchasePrc: Int = 0,
    val totalAmt: Int = 0,
    val returnAmt: Int = 0,
    val transType: TransType,
    val ownedQty: Int = 0,
    val studentId: Long = 0L,
    val stockId: Long = 0L
)

enum class TransType {
    BUY, SELL
}